apps/mobile/src/features/send/send-form/components/send-form-stacks-setter.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { useSendFormContext } from '@/features/send/send-form/send-form-context';
+import { HasChildren } from '@/utils/types';
+import BigNumber from 'bignumber.js';
+import { z } from 'zod';
+
+import { defaultStacksFees, useNextNonce } from '@leather.io/query';
+import { useOnMount } from '@leather.io/ui/native';
+import { microStxToStx } from '@leather.io/utils';
+
+const defaultFeeFallback = 2500;
+
+interface SendFormStacksSetterProps extends HasChildren {
+  address: string;
+}
+export function SendFormStacksSetter({ address, children }: SendFormStacksSetterProps) {
+  const { schema } = useSendFormContext();
+  const { setValue, formState, getValues } = useFormContext<z.infer<typeof schema>>();
+  const { data: nextNonce } = useNextNonce(address);
+
+  // Temporary default fee
+  const defaultFee =
+    defaultStacksFees.estimates[0]?.fee.amount ?? new BigNumber(defaultFeeFallback);
+
```

Comment:
Dislike this inline "setter" component, it's a very side-effecty way of setting state.

What other ways could we do this? Why does it need to be a component? Think it'd be worth trying to find a way to avoid this, as to not copy over bad design patterns from extension.